[BUG] Display freezes on complex web pages (x86) — wlroots DRM atomic commit EBUSY under QtWebEngine GPU load

### Description

On x86 Debian installs using the ghcr.io/screenly/anthias-viewer:latest-x86 image, the display locks on the last rendered frame when the viewer attempts to transition to a complex web page. Simple pages (images, lightweight URLs) transition normally.

Symptom

- anthias-viewer logs cycle through assets normally (Showing asset..., Sleeping for 10)
- AnthiasWebview process remains alive (ps aux confirms it)
- Screen stays frozen on last successfully displayed page
- Restarting the container recovers the display

Root Cause Found in Logs

```
docker logs screenly-anthias-viewer-1 2>&1 | grep -i atomic
[ERROR] [backend/drm/atomic.c:79] connector HDMI-A-1: Atomic commit failed: Device or resource busy
```

The cage wlroots compositor fails its DRM atomic commit because the GPU is still busy completing the QtWebEngine render of the complex page when cage attempts to flip. The kernel returns EBUSY, cage silently gives up, and the screen stays frozen. The Python scheduler has no visibility into this failure.

Workaround

Adding WLR_DRM_NO_ATOMIC=1 to the anthias-viewer container environment resolves the issue by falling back to legacy KMS drmModePageFlip, which is tolerant of GPU fence timing.


```
# docker-compose.override.yml
services:
  anthias-viewer:
    environment:
      - WLR_DRM_NO_ATOMIC=1
      - WLR_DRM_NO_MODIFIERS=1
```

Suggested Fix

Consider adding WLR_DRM_NO_ATOMIC=1 as a default environment variable in docker-compose.yml.tmpl for x86 targets, or detecting the condition in bin/start_viewer.sh and setting it automatically.



### Steps to Reproduce

Steps to reproduce the behavior:
1. Add a complex web site page as an asset, such as <www.marietta.edu>
1. Wait 15-30 min
1. Observe the screen freezes
 
### Expected Behavior

Screen should continue updating.

### Screenshots

If applicable, add screenshots to help explain your problem.

### Environment

- Device: x86 PC (Dell OptiPlex Micro)
- OS: Debian (Trixie/x86)
- Anthias Version: latest-x86 (June 2026)
- Installation Method: curl installer
- Display: HDMI-A-1

Containers Running:

- screenly-anthias-viewer-1   ghcr.io/screenly/anthias-viewer:latest-x86
- screenly-anthias-server-1   ghcr.io/screenly/anthias-server:latest-x86
- screenly-anthias-celery-1   ghcr.io/screenly/anthias-server:latest-x86
- screenly-redis-1            ghcr.io/screenly/anthias-redis:latest-x86

--- a/bin/start_viewer.sh
+++ b/bin/start_viewer.sh
@@ -428,6 +428,31 @@ case "$DEVICE_TYPE" in
     export LIBSEAT_BACKEND=builtin
     export WLR_LIBINPUT_NO_DEVICES=1
 
+    # x86 (Intel i915 in particular) can wedge cage's display pipeline
+    # under heavy QtWebEngine GPU load (issue #2976): wlroots 0.18's
+    # non-blocking atomic page-flip commit gets EBUSY from the kernel
+    # while the previous commit is still fence-blocked behind the web
+    # render, and the trixie cage 0.2 / wlroots 0.18 failure path
+    # doesn't always recover — the screen freezes on the last frame
+    # until the container restarts, even though the viewer keeps
+    # cycling assets. The legacy KMS interface (drmModePageFlip) has
+    # no shallow-queue EBUSY semantics, so it tolerates GPU fence
+    # backpressure; a single-output kiosk uses none of the atomic
+    # extras (VRR, overlay planes, multi-output modeset queueing).
+    # WLR_DRM_NO_MODIFIERS pairs with it: the same trixie wlroots
+    # 0.18.2 + i915 stack has a second documented freeze in the
+    # modifier-fallback swapchain realloc (Debian bug 1112158), and
+    # legacy flips reject scan-out buffers whose modifiers change
+    # between frames — modifier-less allocation keeps the legacy path
+    # maximally reliable. ${VAR:-1} keeps both user-overridable via a
+    # compose override (wlroots parses 0 as false). Scoped to x86:
+    # arm64/pi5 run vc4/rockchip where atomic is the well-exercised
+    # path and no freeze has been observed.
+    if [ "$DEVICE_TYPE" = 'x86' ]; then
+        export WLR_DRM_NO_ATOMIC="${WLR_DRM_NO_ATOMIC:-1}"
+        export WLR_DRM_NO_MODIFIERS="${WLR_DRM_NO_MODIFIERS:-1}"
+    fi
+
     # cage default `-m extend` spans all enumerated DRM outputs,
     # including ones that are physically disconnected — so a Pi user
     # who plugs into the second micro-HDMI port (HDMI-A-2 instead of

